src/Settings.cpp

```diff
@@ -0,0 +1,111 @@
+
+        std::string settingsFileName = currentDir + SETTINGS_FILE_NAME;
+
+        processSettingsFile(settingsFileName);
+    }
+
+    void Settings::processSettingsFile(const std::string& settingsFileName) {
+        try {
+            std::fstream inputFile{ settingsFileName };
+
+            std::cout << "################################################" << std::endl;
+
+            if (!inputFile)
+            {
+                std::cerr << "Could not open the settings file: " << SETTINGS_FILE_NAME << std::endl;
+                std::cerr << "Check if the file was copied correctly at build stage." << std::endl;
+                std::cout << "################################################" << std::endl << std::endl;
+                settingsAreValid = false;
+                return;
+            }
+
+            std::cout << "Settings file opened successfully. Processing..." << std::endl;
+
+            auto jsonFile = web::json::value::parse(inputFile);
+            if (!validateValuesFromDarkSky(jsonFile.at(API_DARKSKY))) {
+                std::cerr << "Invalid Dark Sky API key." << std::endl;
+                std::cout << "################################################" << std::endl << std::endl;
+                settingsAreValid = false;
+                return;
+            }
+
+            this->port_number = jsonFile.at(U("opc_ua_server")).at(U("port-number")).as_integer();
+            this->endpointUrl = utility::conversions::to_utf8string(jsonFile.at(U("opc_ua_server")).at(U("endpoint-url")).as_string());
+            this->hostName = utility::conversions::to_utf8string(jsonFile.at(U("opc_ua_server")).at(U("host-name")).as_string());
+
+            std::cout << "Build completed successfully!!!" << std::endl;
+        }
+        catch (const web::json::json_exception& e) {
+            std::cerr << "Error parsing the settings json file: " << e.what() << std::endl;
+            settingsAreValid = false;
+            return;
+        }
+
+        std::cout << "Weather data units: " << utility::conversions::to_utf8string(units) << std::endl;
+        std::cout << "Interval in minutes for automatic update of weather data: " << intervalWeatherDataDownload << std::endl;
+
+        std::cout << "################################################" << std::endl << std::endl;
+
+        return;
+    }
+
+    bool Settings::validateValuesFromDarkSky(web::json::value& jsonObj) {
+        //Set the values from the Json file's DarkSky object.
+        keyApiDarksky = jsonObj.at(PARAM_NAME_API_DARKSKY_API_KEY).as_string();
+
+        //Simple checks: empty or with spaces - invalid key for sure.
+        if (keyApiDarksky.empty()) {
+            std::cout << "Empty Dark Sky API key. No weather data will be available." << std::endl;
+            return false;
+        }
+        else {
+            for (int i = 0; i < keyApiDarksky.length(); i++) {
+                if (iswspace(keyApiDarksky[i])) {
```

Comment:
Does this work on linux? Because `std::iswspace` expects _wide character_ which is only the case for windows.

Reply:
Just built it on Ubuntu and seems to be OK: no errors and server running just fine.